exercises/verena_reimund/ex02_oo_basics/src/main.cpp

```diff
@@ -8,152 +8,203 @@
 #include <algorithm> //for_each
 //#include <memory>  //smart pointers
 
-// abstract base class
-class shape{
+// abstract base class as interface
+class Shape {// abstract class
 public:
-    shape(){}
-    virtual std::string type() const =0;
-    virtual int number_of_sides() const =0;
-    virtual double perimeter () const =0;
-    virtual double area () const =0;
-    virtual ~shape(){}
+    Shape(){}
+    virtual std::string Type() const =0;
+    virtual int Sides () const =0;
+    virtual double Perimeter () const =0;
+    virtual double Area () const =0;
+    virtual ~Shape(){}
 };
 
 // type definitions
 
-typedef std::vector<shape*> collection; // std::unique_ptr
-typedef std::vector<shape*>::size_type its;
+typedef std::vector<Shape*> collection; // std::unique_ptr
+typedef std::vector<Shape*>::size_type ShapeIterator;
```

Comment:
Slightly unusual name as you `ShapeIterator` is not really an iterator. For example, users might expect after getting a `ShapeIterator` that they might be able to call `begin` and `end` on it.